android/src/main/java/com/tadasr/IOTWifi/IOTWifiModule.java

```diff
@@ -97,9 +104,66 @@ private void connectToWifi(String ssid, String passphrase, Boolean isWEP, Callba
         }
     }
 
+    private void bindToNetwork(final String ssid) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            NetworkRequest.Builder builder = new NetworkRequest.Builder();
+            builder.addTransportType(NetworkCapabilities.TRANSPORT_WIFI);
+            connectivityManager
+                .requestNetwork(builder.build(), new ConnectivityManager.NetworkCallback() {
+                        private boolean bound = false;
+
+                        @Override
+                        public void onAvailable(Network network) {
+                            WifiInfo info = wifiManager.getConnectionInfo();
+                            String offeredSSID = info.getSSID();
+                            if (offeredSSID.startsWith("\"") && offeredSSID.endsWith("\"")) {
+                                offeredSSID = offeredSSID.substring(1, offeredSSID.length() - 1);
+                            }
+
+                            if (!bound && offeredSSID.equals(ssid)) {
+                                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                                    connectivityManager.bindProcessToNetwork(network);
+                                } else {
+                                    ConnectivityManager.setProcessDefaultNetwork(network);
+                                }
+                                try {
+                                    bound = true;
+                                } catch (Exception e) {
+                                    e.printStackTrace();
+                                }
+                            }
+                        }
+
+                        @Override
+                        public void onLost(Network network) {
+                            if (bound) {
+                                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                                    connectivityManager.bindProcessToNetwork(null);
+                                } else {
+                                    ConnectivityManager.setProcessDefaultNetwork(null);
+                                }
```

Comment:
Mind extracting that into one common method taking a `network` param
Above If..else block seems to be repeated from lines 124 - 128 and then 153 - 157.